File: Algorithm/Programmers/Greedy/체육복.java
class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int[] student = new int[n];
        
        for(int i = 0; i<lost.length; i++){
            student[lost[i]-1] = -1;
        }
        
        for(int i = 0; i<reserve.length; i++){
            if(student[reserve[i]-1] == -1){
                student[reserve[i]-1] = 0;
            }else{
                student[reserve[i]-1] = 1;                
            }
        }
        
        for(int i = 0; i<n; i++){
            if(i == 0){
                if(student[i] < 0 && student[i+1] > 0 ){
                    student[i] = 0;
                    student[i+1] = 0;
                }
            }
            else if(i == n-1){
                if(student[i] < 0 && student[i-1] > 0 ){
                    student[i] = 0;
                    student[i-1] = 0;
                }
            }
            else{
                if(student[i] == -1){
                    if(student[i-1] > 0){
                        student[i] = 0;
                        student[i-1] = 0;
                    }else if(student[i+1] > 0){
                        student[i] = 0;
                        student[i+1] = 0;
                    }
                }
            }
        }
        
        int answer = 0;        
        for(int i = 0; i<student.length; i++){
            if(student[i] >= 0){
                answer += 1;
            }
        }
        
        return answer;
    }
}

File: Algorithm/Programmers/Greedy/체육복.js
function solution(n, lost, reserve) {
    var answer = 0;
    let students = new Array(n).fill(1);
    
    //lost = lost.sort((a,b)=>a-b);
    //reserve = reserve.sort((a,b)=>a-b);
    
    let new_lost = lost.filter(val => !reserve.includes(val));
    let new_rv = reserve.filter(val => !lost.includes(val));
 
    for(let l of new_lost){
        students[l-1]-=1;
    }
    
    for(let r of new_rv){
        if(students[r-1]<=0){
            students[r-1]+=1;
            continue
        }
        if(r==1){
            if(students[r]===0){
                students[r]+=1;
            }
        }else if(r == reserve.length){
            if(students[r-2]===0){
                students[r-2]+=1;
            }
        }else{
            if(students[r-2]===0){
                students[r-2]+=1;
            }else if(students[r]===0){
                students[r]+=1;
            }
        }
    }    
    return students.filter((a)=>a).length;
}
